tests/test-grammar-integration.cpp

```diff
@@ -44,21 +44,26 @@ static bool test_build_grammar_fails(const std::string & grammar_str) {
     return grammar_fails;
 }
 
-static bool match_string(const std::string & input, llama_grammar* grammar) {
+static bool match_string(const std::string & input, llama_grammar * grammar) {
     auto decoded = decode_utf8(input, {});
 
     const auto & code_points = decoded.first;
 
+    const llama_grammar_rules  & prev_rules = llama_grammar_get_rules (grammar);
```

Comment:
Nit: For naming convention, no need to prefix with `prev_` anymore, since it's just `rules`.

```suggestion
    const llama_grammar_rules  & rules = llama_grammar_get_rules (grammar);
```